### PR TITLE
fixed some clippy warnings

### PR DESCRIPTION
**Pull request by llogiq**
...and converted some loops to iterators for nicer code (IMHO).

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -131,13 +131,9 @@ impl<T: AsMut<[u8]>> LogBuffer<T> {
...
 
-        let buffer = self.buffer.as_mut();
-        for i in 0..buffer.len() {
-            if is_utf8_leader(buffer[i]) {
-                return core::str::from_utf8(&buffer[i..]).unwrap()
-            }
-        }
-        return ""
+        let buffer = &*self.buffer.as_mut();
+        buffer.iter().cloned().position(is_utf8_leader).map_or("", |i|
```
Comment (whitequark) [inline src/lib.rs:135]: I'm confused. Why did you add `&*`? What's the purpose of `.cloned()` here?

I don't really see how this transformation makes the code more clear...

Comment (llogiq) [reply src/lib.rs:135]: The `&*` is an immutable reborrow because we need to use buffer twice, first for iterating and second for slicing. The `.cloned()` is the canonical way to transform our `Iterator<Item=&u8>` to an `Iterator<Item=u8>`.

Comment (whitequark) [reply src/lib.rs:135]: I see. To be honest I still feel like this change makes the code harder to understand. The code before it was very straightforward.

Your other changes are fine though, I'll cherry-pick them.

**Comment by whitequark** [comment]
Committed in 1c87cfc, thanks!
